diamond.js

```diff
@@ -1,33 +1,52 @@
-let text = {
-    1:" ",2:" ",3:" ",
-    4:" ",5:"*",6:" ",7:" ",8:" ",
-    9:" "
-  };
-  
-  function topDiamond (i,j) {
-  
-    while (i<10 && j>0 && i<=j) {
-          // console.log(i,j);
-      console.log(`${text[1]}${text[2]}${text[3]}${text[4]}${text[5]}${text[6]}${text[7]}${text[8]}${text[9]}`);
-      text[5+i]="*"
-      text[j-5]="*"
-      i +=1;
-      j -=1;
+// PSEUDOCODE HERE
+// 4 blanks 1 asteriks 4 blanks
+// Put a space i times then (length-1)/2 times asteriks then i times space again.
+// We keep decreasing the i number until it hits 0
+// Then we increase it one by one till it's back at 4. 
+
+function topDiamond (width) {
+  let emptySpace=" ";
+  let asterisk="*";
```

Comment:
the static representation which stays the same throughout the code should be define const and globally instead of in functions